sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/NetFrameworkVersion.cs

```diff
@@ -0,0 +1,34 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Helpers
+{
+    public enum NetFrameworkVersion
+    {
+        // either cannot tell, either is is .NET Core or .NET Standard
+        Unknown,
+        // below .NET 4.0
+        Below4,
+        // between .NET 4.0 (inclusive) and .NET 4.5 (exclusive)
+        Between4And45,
+        // after .NET 4.5 (inclusive)
```

Comment:
the APIs have been made secure in 4.5.2, but given that 4.5.2 is the only supported version out of the 4.5* versions, I think it's safe